### ISSUE TITLE
Encode Term sets too many bits

### ISSUE THREAD

**Issue by jorisbontje** [patham9/ANSNA#23]
```
>>SDR Test Start
testing encoding and permutation:
mySDR=
[6](0,6)
[140](1,12)
[280](2,24)
[1311](10,31)
[1312](10,32)
[1313](10,33)
[1314](10,34)
[1315](10,35)
[1316](10,36)
[1317](10,37)
[1318](10,38)
[1319](10,39)
[1320](10,40)
[1321](10,41)
[1322](10,42)
[1323](10,43)
[1324](10,44)
[1325](10,45)
[1326](10,46)
[1327](10,47)
[1328](10,48)
[1329](10,49)
[1330](10,50)
[1331](10,51)
[1332](10,52)
[1333](10,53)
[1334](10,54)
[1335](10,55)
[1336](10,56)
[1337](10,57)
[1338](10,58)
[1339](10,59)
[1340](10,60)
[1341](10,61)
[1342](10,62)
[1343](10,63)
[1344](10,64)
[1345](10,65)
[1346](10,66)
[1347](10,67)
[1348](10,68)
[1349](10,69)
[1350](10,70)
[1351](10,71)
[1352](10,72)
[1353](10,73)
[1354](10,74)
[1355](10,75)
[1356](10,76)
[1357](10,77)
[1358](10,78)
[1359](10,79)
[1360](10,80)
[1361](10,81)
[1362](10,82)
[1363](10,83)
[1364](10,84)
[1365](10,85)
[1366](10,86)
[1367](10,87)
[1368](10,88)
[1369](10,89)
[1370](10,90)
[1371](10,91)
[1372](10,92)
[1373](10,93)
[1374](10,94)
[1375](10,95)
[1376](10,96)
[1377](10,97)
[1378](10,98)
[1379](10,99)
[1380](10,100)
[1381](10,101)
[1382](10,102)
[1383](10,103)
[1384](10,104)
[1385](10,105)
[1386](10,106)
[1387](10,107)
[1388](10,108)
[1389](10,109)
[1390](10,110)
[1391](10,111)
[1392](10,112)
[1393](10,113)
[1394](10,114)
[1395](10,115)
[1396](10,116)
[1397](10,117)
[1398](10,118)
[1399](10,119)
[1400](10,120)
[1401](10,121)
[1402](10,122)
[1403](10,123)
[1404](10,124)
[1405](10,125)
[1406](10,126)
[1407](10,127)
[1561](12,25)
===
SDR term should have TERM_ONES ones
```

The problem is https://github.com/patham9/ANSNA/blob/master/src/Encode.c#L52
```
input_terms[number].blocks[block_i] |= (1 << bit_j);
```

From https://stackoverflow.com/questions/47981/how-do-you-set-clear-and-toggle-a-single-bit:
> Use `1UL` if number is wider than unsigned long; promotion of `1UL << n` doesn't happen until after > evaluating `1UL << n` where it's undefined behaviour to shift by more than the width of a long.

The solution would be:
```
input_terms[number].blocks[block_i] |= (1UL << bit_j);
```

or better / more readable, reuse the SDR bit logic:
```
SDR_WriteBitInBlock(&input_terms[number], block_i, bit_j, 1);
```

**Comment by jorisbontje** [comment]
Similar issue in `SDR_PermuteByRotation` https://github.com/patham9/ANSNA/blob/master/src/SDR.c#L101

Both should be fixed by PR #24

**Comment by patham9** [comment]
Interesting, with my gcc it didn't cause issues. Thank you for fixing!
